src/mame/heathkit/h89.cpp

```diff
@@ -188,6 +207,45 @@ void h89_state::h89_mem(address_map &map)
 	m_mem_view[1](0x1800, 0x1fff).rom().region("maincpu", 0x1800).unmapw();
 }
 
+void h89_state::map_fetch(address_map &map)
+{
+	map(0x0000, 0xffff).lr8(NAME([this](offs_t offset)
+	{
+		return m1_r(offset);
+	}));
+}
+
+uint8_t h89_state::m1_r(offs_t offset)
+{
+	uint8_t data = m_program.read_byte(offset);
+
+	if (m_single_step_enabled && !m_556b_latch)
+	{
```

Comment:
Don’t forget to check `!machine().side_effects_disabled()` or this will cause chaos when debugger windows are open.